### PR TITLE
Flsun V400

### PR DESCRIPTION
**Pull request by dengdaitianhei**
I'm a personnel from flsun ,this is flsun V400 parameter file.Hope to agree as soon as possible

### CODE REVIEW

File: resources/definitions/flsun_v400.def.json [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            "0": "flsun_v400_extruder_0"
+        }
+    },
+    "overrides": {
+        "machine_extruder_count": {
+            "default_value": 1
+        },
+        "retraction_enable": {
+            "default_value": true
+        },
```
Comment (nallath) [inline resources/definitions/flsun_v400.def.json:24]:
```suggestion
```
The default for this setting is already true. No need to set it again.

File: resources/definitions/flsun_v400.def.json [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        "machine_heated_bed": {
+            "default_value": true
+        },
+        "machine_width": {
+            "default_value": 300
+        },
+		"machine_depth": {
```
Comment (nallath) [inline resources/definitions/flsun_v400.def.json:31]:
```suggestion
	"machine_depth": {
```
Fix identation

File: resources/definitions/flsun_v400.def.json [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        },
+        "gantry_height": {
+            "value": "0"
+        },
+        "machine_shape": {
+            "default_value": "elliptic"
+        },
+        "machine_nozzle_size": {
+            "default_value": 0.4
+        },
```
Comment (nallath) [inline resources/definitions/flsun_v400.def.json:59]:
```suggestion
```
That's already the default (so no need to repeat yourself :))

File: resources/definitions/flsun_v400.def.json [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    "metadata": {
+        "visible": true,
+        "author": "FLSUN, Guislain Cyril",
+        "manufacturer": "Flsun",
+        "file_formats": "text/x-gcode",
+		"platform": "flsun_v400.stl",
+        "platform_offset": [0,-40.6455,-43.5],
+        "has_materials": true,
+        "has_machine_quality": true,
+        "machine_extruder_trains":{
```
Comment (nallath) [inline resources/definitions/flsun_v400.def.json:14]:
```suggestion
        "machine_extruder_trains": {
```

**Review by nallath — CHANGES_REQUESTED** [review]

**Review by dengdaitianhei — COMMENTED** [review]
Thank you  for your correction

**Comment by llchuangyi** [comment]
@nallath could add the configuration of v400 in cura as soon as possible

**Comment by antiochsylum** [comment]
Does this have any progress? I see more and more people getting this printer, myself included. What do we need to do to push this through?

**Comment by jellespijker** [comment]
Can you sync your fork, with our `main`. We have added a **printer-linter** tool for our printer definitions and profiles. This should help us with processing our PR's. It performs some standard diagnostic checks and apply auto-formatting on the changed files in this PR, but you can also run it locally. See https://github.com/Ultimaker/Cura/tree/main/printer-linter

If it finds issues for which it knows a fix it should post a comment with that suggested fix such that you can easily commit that fix if you're okay with it.

**Comment by dengdaitianhei** [comment]
> 你好@dengdaitianhei感谢您创建 PR 并将打印机添加到我们的存储库。我们的自动打印机 linter 发现了 PR 的一些问题。该设置`machine_nozzle_size`已经初始化设置为您拥有的相同值。通过在您的设置中覆盖它，没有任何改变。我们的目标是使我们的打印机定义尽可能小，这样做是为了使它们尽可能易于维护。
> 
> 一旦设置从您的定义中删除，我们就可以合并 PR。
> 
> 谢谢

Thanks ,I have deleted the `machine_nozzle_size` setting

File: 资源/定义/flsun_v400.def.json [hunk 5]
```diff
@@ -0,0 +1,70 @@
+{
```
Comment (jellespijker) [inline 资源/定义/flsun_v400.def.json:1]: Please move the printer definition to the `resources/definitions` folder

**Review by jellespijker — CHANGES_REQUESTED** [review]

File: 资源/定义/flsun_v400.def.json (hunk 5 above)
Comment (dengdaitianhei) [reply 资源/定义/flsun_v400.def.json]: That has been completed as required

File: resources/definitions/flsun_v400.def.json [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        "has_machine_quality": true,
+        "machine_extruder_trains": {
+            "0": "flsun_v400_extruder_0"
+        }
+    },
+    "overrides": {
+        "machine_extruder_count": {
+            "default_value": 1
+        },
+        "retraction_enable": {
```
[the remark this answers is not in the document]
Comment (dengdaitianhei) [reply resources/definitions/flsun_v400.def.json]: That has been completed as required

**Review by jellespijker — APPROVED** [review]
LGTM

**Comment by jellespijker** [comment]
Thank you for your contribution, the printer should be available in Cura 5.3

**Pull request merged**